mismi-ec2/src/Mismi/EC2/Data.hs

```diff
@@ -1,65 +1,259 @@
 {-# LANGUAGE OverloadedStrings #-}
 {-# LANGUAGE NoImplicitPrelude #-}
 module Mismi.EC2.Data (
-    InstanceId (..)
-  , UserData (..)
-  , VirtualizationType (..)
-  , SecurityGroupName(..)
-  , SecurityGroupId(..)
-  , renderVirtualization
-  , renderVirtualizationAws
-  , parseVirtualization
+    module Mismi.EC2.Core.Data
+  , fromMismiInstanceType
+  , toMismiInstanceType
+  , fromMismiVirtualizationType
+  , toMismiVirtualizationType
   ) where
 
-import           P
+import qualified Mismi.EC2.Amazonka as A
 
-import           Mismi.EC2.Amazonka (VirtualizationType (..))
+import           Mismi.EC2.Core.Data
 
-newtype InstanceId =
-  InstanceId {
-      instanceId :: Text
-  } deriving (Eq, Show, Ord)
+fromMismiInstanceType :: MismiInstanceType -> A.InstanceType
+fromMismiInstanceType m =
+  case m of
+    C1_Medium ->
+      A.C1_Medium
+    C1_XLarge ->
+      A.C1_XLarge
+    C3_2XLarge ->
+      A.C3_2XLarge
+    C3_4XLarge ->
+      A.C3_4XLarge
+    C3_8XLarge ->
+      A.C3_8XLarge
+    C3_Large ->
+      A.C3_Large
+    C3_XLarge ->
+      A.C3_XLarge
+    C4_2XLarge ->
+      A.C4_2XLarge
+    C4_4XLarge ->
+      A.C4_4XLarge
+    C4_8XLarge ->
+      A.C4_8XLarge
+    C4_Large ->
+      A.C4_Large
+    C4_XLarge ->
+      A.C4_XLarge
+    CC1_4XLarge ->
+      A.CC1_4XLarge
+    CC2_8XLarge ->
+      A.CC2_8XLarge
+    CG1_4XLarge ->
+      A.CG1_4XLarge
+    CR1_8XLarge ->
+      A.CR1_8XLarge
+    D2_2XLarge ->
+      A.D2_2XLarge
+    D2_4XLarge ->
+      A.D2_4XLarge
+    D2_8XLarge ->
+      A.D2_8XLarge
+    D2_XLarge ->
+      A.D2_XLarge
+    G2_2XLarge ->
+      A.G2_2XLarge
+    G2_8XLarge ->
+      A.G2_8XLarge
+    HI1_4XLarge ->
+      A.HI1_4XLarge
+    HS1_8XLarge ->
+      A.HS1_8XLarge
+    I2_2XLarge ->
+      A.I2_2XLarge
+    I2_4XLarge ->
+      A.I2_4XLarge
+    I2_8XLarge ->
+      A.I2_8XLarge
+    I2_XLarge ->
+      A.I2_XLarge
+    M1_Large ->
+      A.M1_Large
+    M1_Medium ->
+      A.M1_Medium
+    M1_Small ->
+      A.M1_Small
+    M1_XLarge ->
+      A.M1_XLarge
+    M2_2XLarge ->
+      A.M2_2XLarge
+    M2_4XLarge ->
+      A.M2_4XLarge
+    M2_XLarge ->
+      A.M2_XLarge
+    M3_2XLarge ->
+      A.M3_2XLarge
+    M3_Large ->
+      A.M3_Large
+    M3_Medium ->
+      A.M3_Medium
+    M3_XLarge ->
+      A.M3_XLarge
+    M4_10XLarge ->
+      A.M4_10XLarge
+    M4_2XLarge ->
+      A.M4_2XLarge
+    M4_4XLarge ->
+      A.M4_4XLarge
+    M4_Large ->
+      A.M4_Large
+    M4_XLarge ->
+      A.M4_XLarge
+    R3_2XLarge ->
+      A.R3_2XLarge
+    R3_4XLarge ->
+      A.R3_4XLarge
+    R3_8XLarge ->
+      A.R3_8XLarge
+    R3_Large ->
+      A.R3_Large
+    R3_XLarge ->
+      A.R3_XLarge
+    T1_Micro ->
+      A.T1_Micro
+    T2_Large ->
+      A.T2_Large
+    T2_Medium ->
+      A.T2_Medium
+    T2_Micro ->
+      A.T2_Micro
+    T2_Nano ->
+      A.T2_Nano
+    T2_Small ->
+      A.T2_Small
```

Comment:
Is this going to have to be bidirectional? As in do you want a mapping the other way. Would be good to add as a test regardless, but feel like you will hit a wall without that mapping as well.